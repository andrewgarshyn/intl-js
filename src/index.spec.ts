/**
 * Copyright 2020, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* eslint-disable no-irregular-whitespace */

import {
  format,
  formatCurrency,
  formatToParts,
  formatCurrencyToParts,
} from '.';

describe('Format', () => {
  describe('number', () => {
    it.each([
      [123456.789, 'PT', '123.456,789'],
      [123456.789, ['CN', 'PT'], '123.456,789'],
    ])('should format %f for %o', (number, locales, expected) => {
      const actual = format(number, locales);
      expect(actual).toBe(expected);
    });
  });

  describe('currency', () => {
    it.each([
      [123456.789, 'PT', '€ 123.456,79'],
      [123456.789, ['CN', 'PT'], '€ 123.456,79'],
    ])('should format %f for %o', (number, locales, expected) => {
      const actual = formatCurrency(number, locales);
      expect(actual).toBe(expected);
    });
  });
});

describe('Format to parts', () => {
  describe('number', () => {
    it.each([
      [
        123456.789,
        'PT',
        [
          { type: 'integer', value: '123' },
          { type: 'group', value: '.' },
          { type: 'integer', value: '456' },
          { type: 'decimal', value: ',' },
          { type: 'fraction', value: '789' },
        ],
      ],
      [
        123456.789,
        ['CN', 'PT'],
        [
          { type: 'integer', value: '123' },
          { type: 'group', value: '.' },
          { type: 'integer', value: '456' },
          { type: 'decimal', value: ',' },
          { type: 'fraction', value: '789' },
        ],
      ],
    ])('should format %f for %o', (number, locales, expected) => {
      const actual = formatToParts(number, locales);
      expect(actual).toEqual(expected);
    });
  });

  describe('currency', () => {
    it.each([
      [
        123456.789,
        'PT',
        [
          { type: 'currency', value: '€' },
          { type: 'literal', value: ' ' },
          { type: 'integer', value: '123' },
          { type: 'group', value: '.' },
          { type: 'integer', value: '456' },
          { type: 'decimal', value: ',' },
          { type: 'fraction', value: '79' },
        ],
      ],
      [
        123456.789,
        ['CN', 'PT'],
        [
          { type: 'currency', value: '€' },
          { type: 'literal', value: ' ' },
          { type: 'integer', value: '123' },
          { type: 'group', value: '.' },
          { type: 'integer', value: '456' },
          { type: 'decimal', value: ',' },
          { type: 'fraction', value: '79' },
        ],
      ],
    ])('should format %f for %o', (number, locales, expected) => {
      const actual = formatCurrencyToParts(number, locales);
      expect(actual).toEqual(expected);
    });
  });
});
