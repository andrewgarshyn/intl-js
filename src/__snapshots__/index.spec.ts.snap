// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Format to parts currency should format 123456.789 for 'PT' 1`] = `
Array [
  Object {
    "type": "currency",
    "value": "€",
  },
  Object {
    "type": "literal",
    "value": " ",
  },
  Object {
    "type": "integer",
    "value": "123",
  },
  Object {
    "type": "group",
    "value": ".",
  },
  Object {
    "type": "integer",
    "value": "456",
  },
  Object {
    "type": "decimal",
    "value": ",",
  },
  Object {
    "type": "fraction",
    "value": "79",
  },
]
`;

exports[`Format to parts currency should format 123456.789 for [ 'CN', 'PT', [length]: 2 ] 1`] = `
Array [
  Object {
    "type": "currency",
    "value": "€",
  },
  Object {
    "type": "literal",
    "value": " ",
  },
  Object {
    "type": "integer",
    "value": "123",
  },
  Object {
    "type": "group",
    "value": ".",
  },
  Object {
    "type": "integer",
    "value": "456",
  },
  Object {
    "type": "decimal",
    "value": ",",
  },
  Object {
    "type": "fraction",
    "value": "79",
  },
]
`;

exports[`Format to parts number should format 123456.789 for 'PT' 1`] = `
Array [
  Object {
    "type": "integer",
    "value": "123",
  },
  Object {
    "type": "group",
    "value": ".",
  },
  Object {
    "type": "integer",
    "value": "456",
  },
  Object {
    "type": "decimal",
    "value": ",",
  },
  Object {
    "type": "fraction",
    "value": "789",
  },
]
`;

exports[`Format to parts number should format 123456.789 for [ 'CN', 'PT', [length]: 2 ] 1`] = `
Array [
  Object {
    "type": "integer",
    "value": "123",
  },
  Object {
    "type": "group",
    "value": ".",
  },
  Object {
    "type": "integer",
    "value": "456",
  },
  Object {
    "type": "decimal",
    "value": ",",
  },
  Object {
    "type": "fraction",
    "value": "789",
  },
]
`;

exports[`Get format currency should get format for 'PT' 1`] = `
Object {
  "currency": "EUR",
  "currencyDisplay": "symbol",
  "currencyIndex": 0,
  "currencySign": "standard",
  "locale": "pt",
  "maximumFractionDigits": 2,
  "minimumFractionDigits": 2,
  "minimumIntegerDigits": 1,
  "notation": "standard",
  "numberingSystem": "latn",
  "parts": Object {
    "currency": "€",
    "decimal": ",",
    "group": ".",
    "literal": " ",
  },
  "signDisplay": "auto",
  "style": "currency",
  "useGrouping": true,
}
`;

exports[`Get format currency should get format for [ 'en-US', 'pt-PT', [length]: 2 ] 1`] = `
Object {
  "currency": "USD",
  "currencyDisplay": "symbol",
  "currencyIndex": 0,
  "currencySign": "standard",
  "locale": "en-US",
  "maximumFractionDigits": 2,
  "minimumFractionDigits": 2,
  "minimumIntegerDigits": 1,
  "notation": "standard",
  "numberingSystem": "latn",
  "parts": Object {
    "currency": "$",
    "decimal": ".",
    "group": ",",
  },
  "signDisplay": "auto",
  "style": "currency",
  "useGrouping": true,
}
`;

exports[`Get format number should get format for 'PT' 1`] = `
Object {
  "locale": "pt",
  "maximumFractionDigits": 3,
  "minimumFractionDigits": 0,
  "minimumIntegerDigits": 1,
  "notation": "standard",
  "numberingSystem": "latn",
  "parts": Object {
    "decimal": ",",
    "group": ".",
  },
  "signDisplay": "auto",
  "style": "decimal",
  "useGrouping": true,
}
`;

exports[`Get format number should get format for [ 'en-US', 'pt-PT', [length]: 2 ] 1`] = `
Object {
  "locale": "en-US",
  "maximumFractionDigits": 3,
  "minimumFractionDigits": 0,
  "minimumIntegerDigits": 1,
  "notation": "standard",
  "numberingSystem": "latn",
  "parts": Object {
    "decimal": ".",
    "group": ",",
  },
  "signDisplay": "auto",
  "style": "decimal",
  "useGrouping": true,
}
`;
